05-html-css/css-grid-intro.md

```diff
@@ -0,0 +1,95 @@
+# Intro to Layout with CSS Grid
+
+## Learning Goals
+
+## History
+
+[Designing on a grid is not a new concept](https://en.wikipedia.org/wiki/Grid_%28graphic_design%29), and in fact grid theory is a well-known practice in many schools of design. However, historically, web designers did not apply these ideas, and CSS did not have available good ways to practice them anyway. It is only until recently that web design has been pushing using a grid for layout.
+
+CSS Grid Layout is a way to determine page layout along a two-dimensional grid (with X and Y axes) with major browser support "out of the box."
+
+CSS Grid has the following advantages:
+1. More applicable than teaching a specific CSS library
+1. Browser compatibility
+1. Promotes responsive design/relative units for layout
+
+Other solutions to layout include:
+- Proprietary implementations of grids in CSS
+  - Bootstrap
+  - Foundation
+- Using combinations of other CSS properties
+  - [Layouts using Display](archived/layout-display.md)
+  - [Layouts using Floats](archived/layout-floats.md)
+  - [Layouts using Positioning](archived/layout-positioning.md)
+
+## Rules for Setting Up CSS Grid
+
+  CSS Grid solves the following problem: **How do I define rules for the layout of many elements, specifically across rows and columns?** Therefore, to use CSS Grid, we set up our CSS Grid with these rules:
+
+  1. There is one large container for these elements, and it is called **the grid container**. We designate it as the grid container by giving it the CSS rule `display: grid;`
+  1. The "many elements" that we are trying to put in a layout are called the grid items. The grid items **must** be and will only be the **direct children** of the grid container
```

Comment:
You add the rule to #1, would it make sense to add the rule to #2 as well?

Reply:
Hm, not sure I understand this: in CSS Grid you only put the rule `display: grid;` in the grid container, so it doesn't apply to the grid items described in number 2.